tracer/src/Datadog.Trace/Agent/MessagePack/SpanMessagePackFormatter.cs

```diff
@@ -189,9 +197,57 @@ private int Serialize(ref byte[] bytes, int offset, in SpanModel spanModel)
             offset += WriteTags(ref bytes, offset, in spanModel, tagProcessors);
             offset += WriteMetrics(ref bytes, offset, in spanModel, tagProcessors);
 
+            if (metaStructValues?.Count > 0)
+            {
+                WriteMetaStruct(ref bytes, ref offset, metaStructValues);
+            }
+
             return offset - originalOffset;
         }
 
+        private void WriteMetaStruct(ref byte[] bytes, ref int offset, IReadOnlyDictionary<string, object> metaStruct)
+        {
+            var data = new Dictionary<string, byte[]>();
+
+            // Since ConcurrentDictionary does not guarantee the order of the keys, we need to order them
+            // to make serialization deterministic.
+            // We need to know the size of the byte array before actually writing it. Depending on the size,
+            // the serializer will use a different amount of bytes in the header to encode the length of the byte array.
+            // That means that we need to know the size of the array, then write the header and then, the array itself.
+
+            foreach (var keyValuePair in metaStruct.OrderBy(keyValuePair => keyValuePair.Key, StringComparer.OrdinalIgnoreCase))
```

Comment:
We should avoid LINQ queries since we can write more efficient code manually. This items probably don't need to be sorted, anyway.

Reply:
LINQ 🙈

Reply:
I have removed the orderBy. It is ok it it's not sorted.

Reply:
Thanks!